realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -183,10 +184,10 @@ public static synchronized void init(Context context) {
             if (context == null) {
                 throw new IllegalArgumentException("Non-null context required.");
             }
-            BaseRealm.applicationContext = context.getApplicationContext();
-            RealmCore.loadLibrary(BaseRealm.applicationContext);
+            RealmCore.loadLibrary(context);
             defaultConfiguration = new RealmConfiguration.Builder().build();
             RealmLog.add(BuildConfig.DEBUG ? new AndroidLogger(Log.DEBUG) : new AndroidLogger(Log.WARN));
+            BaseRealm.applicationContext = context.getApplicationContext();
```

Comment:
Yep, that is possible. We can always refactor if we decide to remove the constructor at some point. It will involve _a lot_ of changes to unit tests though, which was why I kept it outside the scope of this PR